experimenter/experimenter/experiments/models.py

```diff
@@ -127,106 +128,162 @@ class NimbusExperiment(NimbusConstants, TargetingConstants, FilterMixin, models.
+    )
     proposed_duration = models.PositiveIntegerField(
+        "Experiment's expected duration",
         default=NimbusConstants.DEFAULT_PROPOSED_DURATION,
         validators=[MaxValueValidator(NimbusConstants.MAX_DURATION)],
     )
     proposed_enrollment = models.PositiveIntegerField(
+        "Proposed enrollment duration",
         default=NimbusConstants.DEFAULT_PROPOSED_ENROLLMENT,
         validators=[MaxValueValidator(NimbusConstants.MAX_DURATION)],
     )
     population_percent = models.DecimalField[Decimal](
-        max_digits=7, decimal_places=4, default=0.0
+        "Population percent", max_digits=7, decimal_places=4, default=0.0
+    )
+    total_enrolled_clients = models.PositiveIntegerField(
+        "Expected number of clients", default=0
     )
-    total_enrolled_clients = models.PositiveIntegerField(default=0)
     firefox_min_version = models.CharField(
+        "Minimum firefox version",
         max_length=255,
         default=NimbusConstants.Version.NO_VERSION,
         blank=True,
     )
     firefox_max_version = models.CharField(
+        "Maximum firefox version",
         max_length=255,
         default=NimbusConstants.Version.NO_VERSION,
         blank=True,
     )
     application = models.CharField(
+        "Application type",
         max_length=255,
         choices=NimbusConstants.Application.choices,
     )
     channel = models.CharField(
+        "Channel type",
         max_length=255,
         choices=NimbusConstants.Channel.choices,
     )
-    locales = models.ManyToManyField[Locale](Locale, blank=True)
-    countries = models.ManyToManyField[Country](Country, blank=True)
-    languages = models.ManyToManyField[Language](Language, blank=True)
-    is_sticky = models.BooleanField(default=False)
-    projects = models.ManyToManyField[Project](Project, blank=True)
-    hypothesis = models.TextField(default=NimbusConstants.HYPOTHESIS_DEFAULT)
-    primary_outcomes = ArrayField(models.CharField(max_length=255), default=list)
-    secondary_outcomes = ArrayField(models.CharField(max_length=255), default=list)
+    locales = models.ManyToManyField[Locale](
+        Locale, blank=True, verbose_name="Supported locales"
+    )
+    countries = models.ManyToManyField[Country](
+        Country, blank=True, verbose_name="Supported countries"
+    )
+    languages = models.ManyToManyField[Language](
+        Language, blank=True, verbose_name="Supported languages"
+    )
+    is_sticky = models.BooleanField("Sticky enrollment flag", default=False)
+    projects = models.ManyToManyField[Project](
+        Project, blank=True, verbose_name="Supported projects"
+    )
+    hypothesis = models.TextField(
+        "Experiment's hypothesis", default=NimbusConstants.HYPOTHESIS_DEFAULT
+    )
+    primary_outcomes = ArrayField(
+        models.CharField(max_length=255),
+        default=list,
+        verbose_name="Experiment's primary outcomes",
```

Comment:
We don't need `Experiment's` in all these descriptions. Just "Primary Outcomes" is fine.